lib/mindee/parsing/prediction/financial_document/financial_document_v1.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require_relative '../invoice/invoice_v4'
+
+module Mindee
+  module Prediction
+    # Financial document.
+    class FinancialDocumentV1 < InvoiceV4
+      # Time as seen on the receipt in HH:MM format.
+      # @return [Mindee::TextField]
+      attr_reader :time
+      # The receipt category among predefined classes.
+      # @return [Mindee::TextField]
+      attr_reader :category
+      # The receipt sub-category among predefined classes.
+      # @return [Mindee::TextField]
+      attr_reader :subcategory
+      # A classification field, that can return 4 values : 'EXPENSE RECEIPT' ,
+      # 'CREDIT CARD RECEIPT', 'INVOICE', 'CREDIT NOTE'
+      # @return [Mindee::TextField]
+      attr_reader :document_type
+      # Total amount of tip and gratuity. Both typed and handwritten characters are supported.
+      # @return [Mindee::AmountField]
+      attr_reader :tip
+
+      # @param prediction [Hash]
+      # @param page_id [Integer, nil]
+      def initialize(prediction, page_id)
+        # page_id = nil if page_id = 0
+
+        # The API seems to have a typo on these fields, returns them in singular instead of plural
+        prediction['customer_company_registrations'] = prediction['customer_company_registration'] || []
+        prediction['supplier_company_registrations'] = prediction['supplier_company_registration'] || []
+
+        # The API can return this field as nil
+        prediction['supplier_payment_details'] ||= []
+
+        # The API seems to return { 'confidence' => [], 'polygon' => [], 'value' => [] }
+        # instead of { 'confidence' => nil, 'polygon' => [], 'value' => nil }
+        if prediction.dig('tip', 'value').is_a?(Array)
+          prediction['tip'] = { 'confidence' => nil, 'polygon' => [], 'value' => nil }
+        end
+        if prediction.dig('time', 'value').is_a?(Array)
+          prediction['time'] = { 'confidence' => nil, 'polygon' => [], 'value' => nil }
+        end
```

Comment:
Not sure if the API is intended to respond this way or these are possible bugs.

Reply:
This looks like a bug on the API side. I'm checking with the API team.